portals/publisher/src/main/webapp/source/src/app/components/Apis/Create/OpenAPI/Steps/ProvideOpenAPI.jsx

```diff
@@ -95,6 +96,16 @@ export default function ProvideOpenAPI(props) {
         inputsDispatcher({ action: 'isFormValid', value: false });
     }
 
+    function hasJSONStructure (definition) {
+        if (typeof definition !== 'string') return false;
+        try {
+            const result = JSON.parse(definition);
+            return result && typeof result === 'object';
+        } catch (err) {
+            return false;
```

Comment:
We can log the `err` here.
```js
console.log(err);
```
Something like the below can be seen in the console if logged. 

```
VM255:1 Uncaught SyntaxError: Unterminated string in JSON at position 12
    at JSON.parse (<anonymous>)
    at <anonymous>:1:6
 ```

Also, we can use Alert to the user if an error is there.

```js
import Alert from 'AppComponents/Shared/Alert';
.....
Alert.error(`${intl.formatMessage({
                    id: 'your.id',
                    defaultMessage: 'deleted successfully.',
                })}`);
```

Reply:
Added a log with https://github.com/wso2/apim-apps/pull/407/commits/678a9881951b1f99797da5a858d1e97f026cde88. This method is used to check whether the API definition is in JSON format or not. If JSON.parse results in an error it is assumed that the definition is in yaml format